src/neighbors/knn_regressor.rs

```diff
@@ -116,9 +116,9 @@ impl<T: RealNumber, D: Distance<Vec<T>, T>> KNNRegressor<T, D> {
             )));
         }
 
-        if parameters.k <= 1 {
+        if parameters.k < 1 {
```

Comment:
I do not recall why I set it to be strictly > 1. There is no reason why we shouldn't let `k` to be 1